features/search/filters.feature

```diff
@@ -133,6 +133,69 @@ Feature: Filters
       And I should not see "Bilbo Does the Thing"
       And I should not see "Roonal Woozlib and the Ferrets of Nimh"
 
+  @javascript
+  Scenario: Filter a user's bookmarks by bookmarker's tags
+    Given I am logged in as "recengine"
+      And recengine can use the new search
+      And I bookmark the work "Bilbo Does the Thing" with the tags "to read,been here"
+      And I bookmark the work "A Hobbit's Meandering" with the tags "to read"
+      And I bookmark the work "Roonal Woozlib and the Ferrets of Nimh" with the tags "been here"
+
+    # Use an include checkbox
+    When I go to my bookmarks page
+      And I press "Bookmarker's Tags" within "dd.include"
+    Then the "to read (2)" checkbox within "#include_tag_tags" should not be checked
+      And the "been here (2)" checkbox within "#include_tag_tags" should not be checked
+    When I check "to read (2)" within "#include_tag_tags"
+      And I press "Sort and Filter"
+    Then I should see "2 Bookmarks by recengine"
+      And the "to read (2)" checkbox within "#include_tag_tags" should be checked
+      And I should see "Bilbo Does the Thing"
+      And I should see "A Hobbit's Meandering"
+      And I should not see "Roonal Woozlib and the Ferrets of Nimh"
+
+    # Use a second include checkbox for bookmarks with both tags
+    When I check "been here (1)" within "#include_tag_tags"
+      And I press "Sort and Filter"
+    Then I should see "1 Bookmark by recengine"
+      And I should see "Bilbo Does the Thing"
+
+    # Use an exclude checkbox
+    When I go to my bookmarks page
+      And I press "Bookmarker's Tags" within "dd.exclude"
+    Then the "to read (2)" checkbox within "#exclude_tag_tags" should not be checked
```

Comment:
These checkbox steps don't actually do anything:

https://github.com/otwcode/otwarchive/blob/235d5773d24752c4db54f404381141a1880d178d/features/step_definitions/web_steps.rb#L268-L278

However, adding assertions to them reveal failures elsewhere, so I'm leaving that for later. This particular test does pass if the checkbox steps have assertions.

Reply:
[AO3-5380](https://otwarchive.atlassian.net/browse/AO3-5380).